block/submit.go

```diff
@@ -40,22 +41,24 @@ func SubmitLoopInner(
 	ctx context.Context,
 	logger types.Logger,
 	bytesProduced chan int, // a channel of block and commit bytes produced
-	maxBatchSkew uint64, // max number of batches that submitter is allowed to have pending
+	maxBlockSkew uint64, // max number of blocks that submitter is allowed to have pending
+	pendingSubmittedBlocks func() uint64,
 	maxBatchTime time.Duration, // max time to allow between batches
 	maxBatchBytes uint64, // max size of serialised batch in bytes
 	createAndSubmitBatch func(maxSizeBytes uint64) (sizeBlocksCommits uint64, err error),
 ) error {
 	eg, ctx := errgroup.WithContext(ctx)
 
 	pendingBytes := atomic.Uint64{}
+
 	trigger := uchannel.NewNudger()   // used to avoid busy waiting (using cpu) on trigger thread
 	submitter := uchannel.NewNudger() // used to avoid busy waiting (using cpu) on submitter thread
 
 	eg.Go(func() error {
 		// 'trigger': this thread is responsible for waking up the submitter when a new block arrives, and back-pressures the block production loop
 		// if it gets too far ahead.
 		for {
-			if maxBatchSkew*maxBatchBytes < pendingBytes.Load() {
+			if maxBlockSkew <= pendingSubmittedBlocks() {
```

Comment:
As I see it there is an inherent problem with this type of measurement of skew.
the reason being is that we need some way to correlate maxBatchSize with pending blocks. if I set maxBlockSkew to e.g 100, but all the blocks are e.g 1kb, than I get a skew before my batch is considered "full" (if my batch is 500kb).

so basically we're referring to the same issue in 2 different measures that needs to be coordinated, cause batch size defined may be > maxBlockSkew.

The problem with maxBlockSkew is that you can't really predict it's size as it depends on the block size. 

not sure why did you change it from the first place?